src/lib/hostapd.rb

```diff
@@ -9,776 +9,181 @@
 
 module Wifi
 
+  def self.start(config_path, hostapd_path)
+    networks = network_config config_path
+    # TODO: Is this rubyish?
+    if not networks.any?
+      return false
+    end
+    info "found wifis: #{networks.map { |h| h[:name] }}"
+    config_data = create_config_file networks, config_path
+    write_file hostapd_path, config_data
+    # only for debugging purposes:
+    if File.directory? config_path
+      FileUtils.cp hostapd_path, File.join(config_path, 'hostapd.conf.backup')
+    end
+    true
+  end
+
+  private
   def self.wpa_passphrase(ssid, passphrase)
     if [ssid, passphrase].all? { |value| value.to_s.strip.length > 0 }
       OpenSSL::PKCS5.pbkdf2_hmac_sha1(passphrase, ssid, 4096, 32).unpack("H*").first
     end
   end
 
-  module Logging
-    extend Forwardable
-
-    @@logger = Logger.new(STDOUT)
-
-    def self.logger=(logger)
-      @@logger = logger
-    end
-
-    def_delegators :logger, :debug, :info, :warn, :error, :fatal, :unknown
-
-    def log(msg)
-      logger.log
-    end
-
-    def logger
-      @@logger
-    end
-
-
-    def level
-      logger.instance_variable_get(:@level)
-    end
-
-    def level=(level=Logger::ERROR) # DEBUG=0, INFO=1, WARN=2, ERROR=3, FATAL=4, UNKNOWN=5
-      logger.instance_variable_set(:@level, level)
-    end
+  def self.debug msg
+    Logger.new(STDOUT).log Logger::Severity::DEBUG, msg
```

Comment:
Preferrably, introduce a memoized accessor for the logger, this will avoid initializing a new logger object for every log statement

```
def self.logger
  @logger ||= Logger.new(STDOUT)
end

def debug(msg)
  logger.debug 'foo'
end
```